UCS: linking failed due to libiberty.a

I am getting the following error when compiling UCX:

```
make[3]: Entering directory '/home/gaoxiang/torch-ucx-ucc/ucx/src/ucs'
  CCLD     libucs.la
/usr/bin/ld: /usr/lib/gcc/x86_64-pc-linux-gnu/11.2.0/../../../../lib/libiberty.a(rust-demangle.o): warning: relocation against `_sch_istable' in read-only section `.text'
/usr/bin/ld: /usr/lib/gcc/x86_64-pc-linux-gnu/11.2.0/../../../../lib/libiberty.a(cp-demangle.o): relocation R_X86_64_PC32 against symbol `cplus_demangle_builtin_types' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: final link failed: bad value
collect2: error: ld returned 1 exit status
```
The command used to build UCX is:
```bash
rm -rf ucx
git clone --recursive https://github.com/openucx/ucx.git
pushd ucx
./autogen.sh
./configure --prefix=$UCX_HOME      \
    --enable-mt                     \
    --with-cuda=/usr/local/cuda/    \
    --enable-profiling              \
    --enable-stats
make -j
sudo make install
popd
```

--- a/src/ucs/configure.m4
+++ b/src/ucs/configure.m4
@@ -54,11 +54,16 @@ AS_IF([test "x$with_bfd" != xno],
        save_LIBS="$LIBS"
 
        # Check BFD properties with all flags pointing to the custom location
-       CFLAGS="$CFLAGS $BFD_CHECK_CFLAGS"
        CPPFLAGS="$CPPFLAGS $BFD_CHECK_CPPFLAGS"
-       LDFLAGS="$LDFLAGS $BFD_CHECK_LDFLAGS"
        LIBS="$LIBS $BFD_CHECK_LIBS"
        BFD_CHECK_DEPLIBS="-liberty -lz -ldl"
+
+       # Link the test applications as a shared library, to fail if libbfd is
+       # not a PIC object.
+       # Do not allow undefined symbols, to ensure all references are resolved.
+       # TODO Allow static link with static libbfd
+       CFLAGS="$CFLAGS $BFD_CHECK_CFLAGS -fPIC"
+       LDFLAGS="$LDFLAGS $BFD_CHECK_LDFLAGS -shared -Wl,--no-undefined"
 
        bfd_happy="yes"
        AC_CHECK_LIB(bfd, bfd_openr, [],

